### PR TITLE
NSI using the 2.0 API

### PR DESCRIPTION
**Pull request by ajragusa**
There wasn't a ton of work, but it took a long time to work through all the issues... Still need to clean up warns/logs

### CODE REVIEW

File: frontend/webservice/circuit.cgi [hunk 1]
```diff
@@ -125,6 +125,13 @@ my $provision = GRNOC::WebService::Method->new(
     callback    => \&provision,
     description => 'Creates and provisions a new Circuit.'
 );
+$provision->add_input_parameter(
+    name        => 'status',
+    pattern     => '(reserved|confirmed|provisioned|released|decom)',
+    required    => 1,
```
Comment (jonstout) [inline frontend/webservice/circuit.cgi:131]: does this need to be required?

File: frontend/webservice/circuit.cgi [hunk 2]
```diff
@@ -491,6 +499,8 @@ sub update {
     my $add_endpoints = [];
     my $del_endpoints = [];
 
+    warn Dumper($args->{endpoint}->{value});
```
Comment (jonstout) [inline frontend/webservice/circuit.cgi:502]: please remove

File: perl-lib/OESS/lib/OESS/NSI/Processor.pm [hunk 3]
```diff
@@ -99,7 +99,7 @@ sub process_request {
 
     log_info("Received method call: $request");
     log_info(Dumper($request));
-
+    warn Dumper("REceived Method call: $request");
```
Comment (jonstout) [inline perl-lib/OESS/lib/OESS/NSI/Processor.pm:102]: please remove

File: perl-lib/OESS/lib/OESS/NSI/Provisioning.pm [hunk 4]
```diff
@@ -32,7 +32,7 @@ use Data::Dumper;
 use GRNOC::Log;
 use GRNOC::Config;
 use GRNOC::WebService::Client;
-
+use JSON::XS;
```
Comment (jonstout) [inline perl-lib/OESS/lib/OESS/NSI/Provisioning.pm:35]: Not sure if this a new dependency or not. If so we've been using the following elsewhere:
```
use JSON;

my $json = encode_json($obj);
my $obj  = decode_json($json);

File: perl-lib/OESS/lib/OESS/NSI/Provisioning.pm [hunk 5]
```diff
@@ -149,11 +149,16 @@ sub _do_provisioning{
...
-    my $url = $self->{'websvc_location'} . "provisioning.cgi";
+    my $url = $self->{'websvc_location'} . "circuit.cgi";
     $self->{'websvc'}->set_url($url);
 
-    my $res = $self->{'websvc'}->provision_circuit(
-        state => 'provisioned',
+    #my $endpointa = {node => $ckt->{'node'}->[0], interface => $ckt->{'interface'}->[0], tag => $ckt->{'tag'}->[0]};
+    #my $endpointb = {node => $ckt->{'node'}->[1], interface => $ckt->{'interface'}->[1], tag => $ckt->{'tag'}->[1]};
+    #warn Dumper($ckt);
+    #warn "About to provision\n";
```
Comment (jonstout) [inline perl-lib/OESS/lib/OESS/NSI/Provisioning.pm:158]: if we don't need it let's remove it

File: perl-lib/OESS/lib/OESS/NSI/Provisioning.pm [hunk 6]
```diff
@@ -194,62 +199,61 @@ sub _get_circuit_details{
...
-    }
-    
-    my $ckt = { state => $circuit->{'state'},
+#    foreach my $ep (@{$circuit->{'endpoints'}}){
+#	push(@nodes,$ep->{'node'});
+#	push(@ints,$ep->{'interface'});
+#	push(@tags,$ep->{'tag'});
+#    }
+
+    #warn Dumper($circuit);
```
Comment (jonstout) [inline perl-lib/OESS/lib/OESS/NSI/Provisioning.pm:244]: if we don't need it let's remove it

Comment (ajragusa) [reply perl-lib/OESS/lib/OESS/NSI/Provisioning.pm]: All of this should be resolved now

**Pull request merged**